### ISSUE TITLE
Problems in training model with pytorch version code

### ISSUE THREAD

**Issue by mtz1992** [baidut/paq2piq#6]
Excuse me? train.csv How to generate, in the python version of the code, I use the Train it with Pure-Pytorch training model, your model.py Roipoolmodel module y prediction output out in file_ Features are 1-D, but dataset.py The value of Y in the file is label_ cols = 'mos_ image', 'mos_ patch_ 1', 'mos_ patch_ 2', 'mos_ patch_ 3 ', which means 4 dimensions.Can you give a detailed description of the training process and configuration parameters using pytorch?
Second, what is the input of X when training with the pytorch version code, the whole picture, or three patches, or input together

**Comment by baidut** [comment]
Sorry for the late reply.  Thank you for posting the problem.

The model was trained using FastAI. In this pure pytorch version we did provide code for training but the performance are not guaranteed, because the current implementation is not exactly the same with the FastAI version.

Please check out the label files here: https://github.com/baidut/PaQ-2-PiQ/tree/master/database

* `labels=640_padded.csv` training + validation set
* `labels640.csv`  test set

**Comment by mtz1992** [comment]
Thank you very much.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Zhenqiang.Ying"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月11日(星期四) 上午9:39
收件人:&nbsp;"baidut/paq2piq"<paq2piq@noreply.github.com&gt;;
抄送:&nbsp;"漫步云端"<a75468916@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [baidut/paq2piq] Problems in training model with pytorch version code (#6)





 
Sorry for the late reply.  Thank you for posting the problem.
 
The model was trained using FastAI. In this pure pytorch version we did provide code for training but the performance are not guaranteed, because the current implementation is not exactly the same with the FastAI version.
 
Please check out the label files here: https://github.com/baidut/PaQ-2-PiQ/tree/master/database
  
labels=640_padded.csv training + validation set
 
labels640.csv  test set

**Comment by mtz1992** [comment]
So, if I use the fastai source code, 

can I train like this? If I can, then I can train bestmodel.pth File is unpredictable? Can you send me the code corresponding to the training and prediction?Please.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Zhenqiang.Ying"<notifications@github.com&gt;;
发送时间:&nbsp;2020年6月11日(星期四) 上午9:39
收件人:&nbsp;"baidut/paq2piq"<paq2piq@noreply.github.com&gt;;
抄送:&nbsp;"漫步云端"<a75468916@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [baidut/paq2piq] Problems in training model with pytorch version code (#6)





 
Sorry for the late reply.  Thank you for posting the problem.
 
The model was trained using FastAI. In this pure pytorch version we did provide code for training but the performance are not guaranteed, because the current implementation is not exactly the same with the FastAI version.
 
Please check out the label files here: https://github.com/baidut/PaQ-2-PiQ/tree/master/database
  
labels=640_padded.csv training + validation set
 
labels640.csv  test set
